ClientApp/src/components/AudioPlayer.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { Button } from "semantic-ui-react";
+import "./AudioPlayer.css";
+//Instal react icon library by running 'npm install react-icons' in terminal
+import {GrForwardTen} from "react-icons/gr";
+import {GrBackTen} from "react-icons/gr";
+import { AiFillPauseCircle } from "react-icons/ai";
+import { AiFillPlayCircle } from "react-icons/ai";
+
+
+const AudioPlayer = ({src}) => {
+  //state
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [duration, setDuration] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+
+  //references
+  const audioPlayer = useRef(); //reference audio component
+  const progressBar = useRef(); //reference progress bar component
+  const animationRef = useRef(); //reference the animation
+
+  //effect
+  useEffect(() => {
+    const seconds = Math.floor(audioPlayer.current.duration);
+    setDuration(seconds);
+    progressBar.current.max = seconds;
+  }, [audioPlayer?.current?.loadedmetadata, audioPlayer?.current?.readyState]);
+
+
+  const calculateTime = (secs) => {
+    const minutes = Math.floor(secs/60);
+    const returnMinutes = minutes < 10 ? `0${minutes}` : `${minutes}`;
+    const seconds = Math.floor(secs % 60);
+    const returnSeconds = seconds < 10 ? `0${seconds}` : `${seconds}`;
+    return `${returnMinutes}:${returnSeconds}`; 
+  }
+
+
+  const togglePlayPause = () => {
+    const prevValue = isPlaying;
+    setIsPlaying(!prevValue);
+    if (!prevValue) {
+      audioPlayer.current.play();
+      animationRef.current = requestAnimationFrame(whilePlaying);
+    } else {
+      audioPlayer.current.pause();
+      cancelAnimationFrame(animationRef.current);
+    }
+  }
+
+  const whilePlaying = () => {
+    progressBar.current.value = audioPlayer.current.currentTime;
+    changePlayerCurrentTime();
+    animationRef.current = requestAnimationFrame(whilePlaying);
+  }
+
+  const changeRange = () => {
+    audioPlayer.current.currentTime = progressBar.current.value;
+    changePlayerCurrentTime();
+  }
+
+  const changePlayerCurrentTime = () => {
+    progressBar.current.style.setProperty('--seek-before-width', `${progressBar.current.value / duration * 100}%`)
```

Comment:
This is the reason you were getting a `NaN` at first.
`duration` is populated later when you hit play and the file is downloaded. So the fix is to add a check `duration != 0` in this line. And set the value of the property to zero if that's the case.

Reply:
Okay I’ll do that